pkg/collector/adapters/config_to_ports.go

```diff
@@ -60,28 +61,34 @@ func ConfigToReceiverPorts(logger logr.Logger, config map[interface{}]interface{
 		return nil, ErrReceiversNotAMap
 	}
 
+	sortedNames := make([]string, 0, len(receivers))
```

Comment:
@kevinearls wouldn't it be cleaner to sort `ports` at the end of the function?

Reply:
Maybe, but I'm not sure it necessary.  Originally when I saw this problem it occurred because we were getting receivers in the wrong order.  Sorting them alphabetically by name is fairly simple.

The ports, on the other hand, are in an array of corev1.ServicePort.  That does not implement methods needed to feed it to sort.Sort(), so sorting those would be non-trivial.

Reply:
What is the root cause of the problem? That the ports are not sorted or receivers? We should use the least error-prone approach:


Take a look at `sort.Slice`:

```
	sort.Slice(ports, func(i, j int) bool {
		return ports[i].Name > ports[j].Name
	})
```

Reply:
The problem I had was caused by the receivers being in the wrong order, not the ports for a receiver.